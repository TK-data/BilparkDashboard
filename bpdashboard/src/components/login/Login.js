import React from 'react';
import './../../styles/login.css';


class Login extends React.Component {
  render(){
    return(
      <form className="loginContainer">
        <input className="userInput" type="text" placeholder="Brukernavn" required="required" />
        <input className="userInput" type="password" placeholder="Passord" required="required" />
        <button className="loginButton"><div className="buttonText">Logg inn</div></button>
      </form>
    )
  }
}
export default Login;
