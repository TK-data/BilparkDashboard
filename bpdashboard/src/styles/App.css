.App {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 70vh;
}

.App-logo {
  width: 300px;
  animation:
    driving 1.9s infinite;
}

.App-header {
  display: flex;
  background-color: rgb(000,039,118);
  padding: 20px;
  color: white;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
  color: white;
}


@keyframes driving {
  0% {
    transform: translateY(0) rotate(0deg);
  }
  10% {
    transform: translateY(-4px) rotate(-1deg);
  }
  20% {
    transform: translateY(0) rotate(0deg);
  }
  30%{
    transform: translateY(-4px) rotate(0deg);
  }
  40%{
    transform: translateY(0) rotate(0deg);
  }
  50%{
    transform: translateY(-4px) rotate(0deg);
  }
  60%{
    transform: translateY(0) rotate(0deg);
  }
  70%{
    transform: translateY(-4px) rotate(1deg);
  }
  80%{
    transform: translateY(0) rotate(0deg);
  }
  90%{
    transform: translateY(-4px) rotate(0deg);
  }
  100%{
    transform: translateY(0) rotate(0deg);
  }
}
