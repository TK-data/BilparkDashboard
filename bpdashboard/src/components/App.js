import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import Login from './login/Login';
import LoginTest from './login2/Logintest';
import logo from './car.png';
import './../styles/App.css';
class App extends Component {

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Velkommen til Bilparken!</h1>
        </header>
        <div className="App-intro">
        </div>
        <main>
          <Route exact path="/" component={Login} />
          <Route exact path="/admin" component={LoginTest} />
        </main>
      </div>
    );
  }
}

export default App;
