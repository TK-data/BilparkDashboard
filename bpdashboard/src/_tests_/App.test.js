import React from 'react';
import ReactDOM from 'react-dom';
import App from './../components/App';
import Login from '../components/login/Login';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../store';

describe('AppComponent', () => {
  test('should render', () => {
    const component = renderer
      .create(
        <Provider store={store}>
          <MemoryRouter>
            <App />
          </MemoryRouter>
        </Provider>,
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
