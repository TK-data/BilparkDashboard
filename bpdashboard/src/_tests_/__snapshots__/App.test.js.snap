// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should render 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="car.png"
    />
    <h1
      className="App-title"
    >
      Velkommen til Bilparken!
    </h1>
  </header>
  <div
    className="App-intro"
  />
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/about-us"
      onClick={[Function]}
    >
      About
    </a>
  </header>
  <main>
    <form
      className="loginContainer"
      name="login"
      onSubmit={[Function]}
    >
      <input
        className="userInput"
        placeholder="Brukernavn"
        required="required"
        type="text"
      />
      <input
        className="userInput"
        placeholder="Passord"
        required="required"
        type="password"
      />
      <button
        className="loginButton"
      >
        <div
          className="buttonText"
        >
          Logg inn
        </div>
      </button>
    </form>
  </main>
</div>
`;
